packages/opentelemetry-sdk-trace-base/src/export/BatchSpanProcessorBase.ts

```diff
@@ -173,8 +174,21 @@ export abstract class BatchSpanProcessorBase<T extends BufferConfig>
                   new Error('BatchSpanProcessor: span export failed')
               );
             }
-          }
-        );
+          });
+        const pendingResources = spans
+          .map(span => span.resource)
+          .filter(resource => !resource.asyncAttributesHaveResolved());
```

Comment:
now that I think about it again, maybe the user starts multiple tracer providers, each with a different resource, and connects them all to the same processor? In this case, different spans might have different resources which we will have to iterate.

Reply:
I'm not sure if this is actually a supported setup. As far as I remember the OTEL protocol sends resource once per message not once per span.
I think this resource awaiting should be triggered only once on first `onEnd()` instead iterating all spans again and again.

Reply:
Yes correct. Can't remember where I saw this single resource per export thingy.
Anyhow, more `TracerProvider` in a single process is likely a fun thing as only one of them can be global but on default all of them refer to the global `ContextManager`...

Reply:
> Anyhow, more `TracerProvider` in a single process is likely a fun thing as only one of them can be global but on default all of them refer to the global `ContextManager`...

Yeah, this will cause all kinds of issues but is technically possible.
I personally think that it's better to just await a single resource and remove the arrays and iterations, but can understand if other people prefer the safer less performant option.

Reply:
Since this is technically possible, should I keep the code as is, or remove the iterations on the spans?

Reply:
I've been looking through the spec and don't see anywhere that prevents multiple resources from being sent to the same batch processor. Although unlikely, I think we have to assume it may happen unless and until specified otherwise